### PR TITLE
replace uprn with locationId

### PR DESCRIPTION
**Pull request by elena-vi**

### CODE REVIEW

**Comment by elena-vi** [comment]
[🐛 Bug fix: fix how time slots are displayed on availability page](https://trello.com/c/6o6fBMph/276-%F0%9F%90%9B-bug-fix-fix-how-time-slots-are-displayed-on-availability-page)

File: compoments/report-repair/repair-availability.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const RepairAvailability = ({handleChange, values, nextAvailability}) => {
     repairLocation:  values.repairLocation,
     repairProblem:  values.repairProblem,
     repairIssue: values.repairProblemBestDescription,
-    uprn: values.uprn,
+    locationId: values.address,
```
Comment (Bhavesh-Patel) [inline compoments/report-repair/repair-availability.js:17]: I think `values.address` won't be the `locationId` required by the API - it's value is currently the lines of the address.

image

Have I missed the change that sets `values.address` to an ID?

I believe that the `Address` component needs to be updated to store the id and the address value in the state.

Comment (Bhavesh-Patel) [reply compoments/report-repair/repair-availability.js:17]: After updating the API to return `locationId` the value of address is now prefixed with the locationId:
image

...it should be separated to 2 distinct properties of the `address` value in the state.

File: compoments/report-repair/repair-availability.js [hunk 2]
```diff
@@ -8,13 +8,12 @@ import moment from 'moment';
...
   const [error, setError] = useState();
   const [value, setValue] = useState();
-  console.log(values)
   const baseURL = '/api/availability';
   const params =  {
     repairLocation:  values.repairLocation,
     repairProblem:  values.repairProblem,
     repairIssue: values.repairProblemBestDescription,
-    locationId: values.address,
+    locationId: values.address?.locationId,
```
Comment (Bhavesh-Patel) [inline compoments/report-repair/repair-availability.js:16]: Could `values.address` be null?
If so, given `locationId` is a required parameter (as are the others), then we should avoid calling the API when any required parameter is not set.

**Review by Bhavesh-Patel — APPROVED** [review]
In relation to my comment, validating API call parameters would have wider scope beyond this PR.

File: compoments/report-repair/repair-availability.js (hunk 2 above)
Comment (elena-vi) [reply compoments/report-repair/repair-availability.js:16]: this was more to the page could render without data for testing and local dev

**Pull request merged**